rust/catalyst-types/src/id_uri/mod.rs

```diff
@@ -329,18 +329,18 @@ impl IdUri {
     /// # Examples
     ///
     /// ```rust
-    /// use catalyst_types::id_uri::{role_index::RoleIndex, IdUri};
+    /// use catalyst_types::id_uri::{role_index::RoleId, IdUri};
     ///
     /// let id_uri = "id.catalyst://cardano/FftxFnOrj2qmTuB2oZG2v0YEWJfKvQ9Gg8AgNAhDsKE"
     ///     .parse::<IdUri>()
     ///     .unwrap();
-    /// let new_role: RoleIndex = 5.into();
+    /// let new_role = RoleId::try_from(3).unwrap();
```

Comment:
Do we really need to use try_from/unwrap?
I am ok with saturating at some large unrealistic value, and it shouldn't fail here if its outside the bounds of the enum itself, as the enum is non exhaustive?

A value outside our known enums is not an invalid value, we just don't know what role that is and would process any value outside our known values as "unknown" role.